<!DOCTYPE html>
<html>

<head>
    <!-- Compiled and minified CSS -->
    <link rel="stylesheet" href="/assets/materialize.min.css">
    <link rel="stylesheet" href="index.css">
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
    <link href="all.css" rel="stylesheet">
    <!-- Compiled and minified JavaScript -->
    <script src="/assets/materialize.min.js" type="text/javascript"></script>
</head>
<div class="navbar-fixed">
    <nav>
        <div class="nav-wrapper">
            <a href="/tutorial" class="brand-logo center">MAD Tutorial</a>
            <ul id="nav-mobile" class="left hide-on-med-and-down">
                <li><a href="database.html">Database</a></li>
                <li><a href="molecule.html">Molecule Builder</a></li>
                <li><a href="system.html">System Builder</a></li>
                <li class="active"><a href="polymer.html">Polymer generator</a></li>
                <li><a href="api.html">API</a></li>
                <li><a href="architecture.html">Architecture</a></li>
            </ul>
        </div>
    </nav>
</div>

<div class="container">
    <h1>Polymer generator</h1>
    <div id="header">
        Welcome to our new MAD feature! We are excited to offer you an easy way to generate new polymer (for example:
        polysaccharides, Battery polymer) or to modify existing molecules (for example Protein PEGylation,
        Glycosylation). This interface allows anyone with a free account to create new molecules in a user-friendly way, without any
        command line or computing skills. Our interface is fully automated, using <a
            href="https://github.com/marrink-lab/polyply_1.0">
            polyply</a>[<a href="#references">1</a>] program to generate the
        necessary files and coordinates for molecular dynamic simulations. Users can generate a molecule using some of
        the molecular compounds provided by Polyply by using the Martini3 forcefield or the Martini 2 forcefield, or you can load your molecule
        (from the MAD database, MAD history, or upload file directly). 
    </div>

    <div id="summary">
        <p>
            The following example will guide you through the multiple uses of <span class="MAD_logo">MAD: Polymer
                generator</span>.
        </p>
        <h6> <a href="#organization">1. How does it work? </a></h6>
        <h6> <a href="#simple_peptide"> 2. Create a simple peptide </a> </h6>
        <h6> <a href="#isopeptidique_link"> 3. Create an isopeptide link </a> </h6>
        <h6> <a href="#load">4. Load a molecule</a> </h6>
        <h6> <a href="#pegylation"> 5. Pegylate a protein </a> </h6>
        <h6> <a href="#limitation">6. Limitation </a></h6>
    </div>

    <div class="disclaimer" style="width: 75%;margin-bottom: 2em">
        <div><i class="medium material-icons inline-icon">person_add</i></div>
        <div>
            Because the service makes use of our computational resources, it is required for you to <a
                href="/login">sign
                in</a> order to have access to the <span class="MAD_logo">MAD:System Builder</span>
        </div>
    </div>


    <div id="organization">
        <h4>General organization
        </h4>


        <div>
            The client interface is divided into two sections: the viewer and the menu.
            The viewer allows users to interact with their molecule through a custom right-click panel menu, which
            changes based on the location clicked (a molecule, a link, or nowhere) and provides a feature (remove
            residue,
            select the whole polymer, remove the link, and more).
            The menu allows users to add new residues, load molecules, generate links, and, once the molecule is
            complete, send it to the server for file generation. The server then runs the Polyply program to generate
            the topology and coordinates. If there are any issues with the process, an error message is displayed to the
            user, providing more accurate information with the involvement of Polyply. Once the coordinates are
            generated, a small minimization is run with <a href="https://www.gromacs.org/about.html">gromacs </a> to
            ensure the reliability of the molecule, which is then displayed in the client interface. Users have the
            option to download all of the generated files and add them to their
            history for future use with other MAD features.
            
           
            <br>
            <div style="color: red;">
                To be more precise, when we use martini2 version, it is the version martini_v2.1-dna.itp and for martini3 is martini_v3.0.0.itp.

                <br>
                Ajouter un schema
            </div>

        </div>
    </div>
    <br>

    <div id="simple_peptide">
        <h4>Create a simple peptide</h4>
        <div class="flex-container">
            <div class="comment">

                <div>
                    In this part, we generate a small peptide of 5 amino acids composed of 3 GLU and 2 ASP with martini3 forcefield.
                </div>

                <div class="main_comment">
                    <ul class="show">
                        <li>
                            <span class="li_supt">Add tripeptide:</span> Select the forcefield martini3 and then choose the good amino acid (here we choose
                            Glutamic acid) and select the number of residues that you want, in our case we choose 3 repetitions.
                            So a chain of 3 Glutamic acids will be added.
                        </li>
                        <li>
                            <span class="li_supt">Add dipeptide and link:</span> Add another amino acid chain ( 2
                            Arginines) and link, then to our previous molecule (you can drag and drop with your mouse to create a link,
                            use the menu, or click right and use the "magic link")
                        </li>
                        <li>
                            <span class="li_supt">Send for computation:</span> Send to the server to generation, you can
                            change the
                            size of the box for coordinate generation, change the number of molecules that you want in
                            this box and most important choose a name for your molecule.
                        </li>
                        <li>
                            <span class="li_supt">Done:</span> You can download your molecule and save it in your
                            history.
                        </li>
                    </ul>
                </div>
            </div>
            <div>
                <img src="images/polymer/peptide.gif" style="width: 700px;" />
            </div>
        </div>
        <div class="info">
            <i class="material-icons inline-icon">info_outline</i>
            Be sure that you have linked residue with a continuous id number ( do not do : (1-2-3)-(7-8)-(4-5-6) do
            (1-2-3)-(4-5-6)-(7-8) instead). It may generate an error with polyply if the continuity is not respected.
        </div>

    </div>

    <br>


    <div id="load">
        <h4>How to load a molecule</h4>

        <div>
            You can load and import molecules in different ways.
        </div>

        <div class="flex-container">
            <div class="comment">
                <div class="main_comment">
                    <ul>
                        <li>
                            <span class="li_supt">Upload a simple molecule :</span> You can load serveral type of files
                            in Polymer Generator.Here are the different types of files you can upload (in these cases,
                            you don't have any
                            information about the coordinate).
                            <ul style="padding-left: 5%;">
                                <li><span class="li_supt">JSON file</span>, who contains a polymer description that you
                                    have
                                    previoulsy download in JSON format or that you have allready created with polyply
                                    "gen-itp".This JSON should contain a list of nodes (molecules) and links between
                                    them. Exemple <a href="/public/tutorial/files/exemple.json"> here</a>.
                                </li>
                                <li>
                                    <span class="li_supt">Fasta file</span>: A protein sequence in Fasta format who
                                    respect the one letter <a href="https://www.bioinformatics.org/sms/iupac.html">
                                        code</a>.
                                </li>
                                <li>
                                    <span class="li_supt">ITP file</span>: A topology file for a molecule that is not
                                    already inside polyply database, in this case, be sure that your molecule is right
                                    coarse-grained in martini3.
                                </li>
                            </ul>
                        </li>
                        <li>
                            <span class="li_supt">Modify a molecule (Add/substitute polymer to a macromolecule):</span>
                            To load your molecule and add/substitute polymer you need to provide 2 different files :
                            coordinates (format .gro) and the topology (format .itp).
                        </li>
                        <li>
                            <span class="li_supt">Load from database or history (Advanced fonction)</span> You can load
                            a
                            molecule from the database or your history. You will load topology and coordinate
                            file.
                        </li>
                        <li>
                            <span class="li_supt">Design your own itp link file (Avanced fonction)</span> Because
                            polyply has a limited list of a molecule, you would want to create your linker file to
                            have better control of the output result. You can see how to design this file <a
                                href="https://github.com/marrink-lab/polyply_1.0/wiki/Tutorial:-writing-.ff-input-files">here</a>.
                        </li>
                        </li>
                    </ul>
                </div>
                <div class="warning">
                    <i class="material-icons inline-icon">warning</i>
                    Impossible to load 2 coordinates files for now.
                    <span style="font-weight: initial;">In such case, the first coordinate file will be replaced by the
                        new one.</span>

                </div>

            </div>


        </div>
        <br>
        <div>
            <img src="images/polymer/load.png" style="width: 1000px;" />
        </div>

    </div>

    <br>

    <div id="isopeptidique_link">
        <h4>Create an isopeptidique link</h4>
        <div class="flex-container">
            <div class="comment">

                <div>
                    In this part, we want to create a link not handled by polyply. We want to do an isopeptide link
                    between a peptide that contains only lysine for instance and a peptide that contains valine and
                    Glutamic
                    acid. In this case, we don't have to change the beads type.
                </div>

                <div class="main_comment">
                    <ul class="show">
                        <li>
                            <span class="li_supt">Create 2 peptides:</span> As we have seen before, create 2 peptide
                            chains. The first one will contain only Valine and the second one contains different amino
                            acids
                        </li>
                        <li>
                            <span class="li_supt">Link and send:</span> Now you can link it and send it to the server
                            to computation.
                        </li>
                        <li>
                            <span class="li_supt">Fix link:</span> A problem with be raised by polyply. The link between
                            amino acid is not provided by the polyply database. So you should fix this link, you can
                            define which bead you want to connect, change beads type, and other link parameters.
                        </li>

                    </ul>
                </div>
            </div>
            <div>
                <img src="images/polymer/isopeptidique_link.gif" style="width: 700px;" />
            </div>
        </div>


    </div>




    <br>
    <div id="pegylation">
        <h4>Pegylate a protein</h4>


        <div>
            In this part, we want to apply a post-translation modification : attaching polyethylene glycol (PEG or PEO)
            molecules to a protein. This modification is called Pegylation and it helps improve the properties
            of a protein :
            <ul style="margin-left: 100px;">
                <li style="margin-bottom: 3px;">
                    Stability: Polyethylene glycol is a stable molecule that does not break down easily, so attaching it
                    to a protein can help to stabilize the protein and extend its shelf life.
                </li>
                <li style="margin-bottom: 3px;">
                    Solubility: This molecule is hydrophilic (water-loving), so attaching it to a protein can
                    help to increase the protein's solubility in water. This can be especially useful for
                    proteins that tend to aggregate or clump together, as the PEG can help to keep them in solution.
                </li>
                <li style="margin-bottom: 3px;">
                    Lifetime: Polyethylene glycol can shield a protein from being recognized and broken down by the
                    body's immune system, which can help to extend the protein's half-life in the body. This can be
                    especially
                    useful for proteins that are normally broken down quickly, as it allows them to remain
                    active for a longer period.
                </li>
                <li style="margin-bottom: 3px;">Targeting: It can be attached to proteins in such a way that it can act
                    as a "stealth" molecule, helping the protein to evade the body's immune system and targeting
                    specific cells
                    or tissues for delivery.
                </li>
            </ul>


        </div>

        <br>
        <div class="flex-container">
            <div class="comment">

                <div class="main_comment">
                    <ul class="show">
                        <li>
                            <span class="li_supt">Load your protein:</span> Load the file about your protein: we need
                            the
                            coordinates of the protein (format .gro) and the topology (format .itp). Section "Modify a
                            protein" and upload your files.
                        </li>
                        <li>
                            <span class="li_supt">Add polyethylene glycol:</span> Then add your polyethylene glycol
                            (called PEO on the polyply molecule list residue). You can choose to link them directly on a
                            certain type of
                            residue (Alanine for instance), then you will add a chain of PEO of the desired length to
                            every Alanine. You can also do it by hand: generate a chain and plug it into your protein
                            "manually".
                        </li>
                        <li>
                            <span class="li_supt">Send for computation :</span> Send to the server to do your
                            modification. You will have a mistake because polyply doesn't know how to link a PEO and an
                            amino acid. So click on the fixed link and select the beads that you want to connect (You
                            can apply to every residue of the
                            same type).
                        </li>


                        You can download the files used for the demo :
                        <ul class="show">
                            <li style="margin-bottom: 3px;">
                                Topology <a href="/files/topology.itp" download>
                                    file</a>
                            </li>
                            <li style="margin-bottom: 3px;">
                                Coordinate <a href="/files/coordinate.gro" download>
                                    file
                                </a>
                            </li>
                        </ul>
                </div>
            </div>
            <div>
                <img src="images/polymer/peg.gif" style="width: 700px;" />
            </div>
        </div>
    </div>
    <br>



    <br>

    <div id="limitation">
        <h4>Current limitation</h4>

        <div style="color: red;">
            <div>
                There is still some limitation to this new MAD feature :
                <ul style="margin-left: 50px;">
                    <li>
                        <span class="li_supt">Impossible generate structure:</span>, For now, generate big structure is not avaible. Because computational resources are limited and not allowed to generate a big system. For exemple it's impossible to create a sphere of a cylinder.
                    </li>
           
                    <li>
                        <span class="li_supt">Number of molecule limited:</span> Polyply has a limited number of
                        molecule
                        avaible. More molecules will be available in future updates. You can find all the molecules
                        containing
                        in polyply database for martini3 <a
                            href=" https://github.com/marrink-lab/polyply_1.0/tree/master/polyply/data/martini3">
                            here</a>.

                    </li>
                    <li>
                        <span class="li_supt"> Links are limited:</span> For the moment all links between molecules are
                        provided by polyply...
                    </li>
                </ul>


                ... Add chirality
                ...blend of polymer

            </div>
        </div>
    </div>


    <div id="references">
        <h4>References</h4>
        <ol>
            <li>
                Grünewald, F., Alessandri, R., Kroon, P.C. et al. Polyply; a python suite for facilitating
                simulations
                of
                macromolecules and nanomaterials. Nat Commun 13, 68 (2022).
                https://doi.org/10.1038/s41467-021-27627-4
            </li>
        </ol>
    </div>

</div>

</html>